src/filebrowser/dialogs.ts

```diff
@@ -75,6 +93,33 @@ function createNewDialog(model: FileBrowserModel, manager: DocumentManager, host
 
 
 /**
+ * Rename a file with optional dialog.
+ */
+export
+function renameFile(model: FileBrowserModel, oldPath: string, newPath: string): Promise<IContents.IModel> {
+  return model.rename(oldPath, newPath).catch(error => {
+    if (error.xhr) {
+      error.message = `${error.xhr.statusText} ${error.xhr.status}`;
+    }
+    if (error.message.indexOf('409') !== -1) {
+      let options = {
+        title: 'Overwrite file?',
+        body: `"${newPath}" already exists, overwrite?`,
+        okText: 'OVERWRITE'
+      };
+      return showDialog(options).then(button => {
+        if (button.text === 'OVERWRITE') {
+          return model.deleteFile(newPath).then(() => {
+            return model.rename(oldPath, newPath);
+          });
+        }
+      });
+    }
+  });
```

Comment:
This `catch` handler should rethrow the error if it fails to overwrite or correct the error condition. Otherwise, the error will be lost in the ether and the UI will do nothing. That will be hard to debug. Audit the exit points here and make sure you throw something.